spdlog/src/formatter/pattern_formatter/mod.rs

```diff
@@ -0,0 +1,1446 @@
+/// | --- | --- | --- |
+/// | `{a}`, `{weekday-name}` | Abbreviated weekday name | `Mon`, `Tue` |
+/// | `{A}`, `{weekday-name-full}` | Weekday name | `Monday`, `Tuesday` |
+/// | `{b}`, `{month-name}` | Abbreviated month name | `Jan`, `Feb` |
+/// | `{B}`, `{month-name-full}` | Month name | `January`, `February` |
+/// | `{c}`, `{datetime}` | Full date time | `Thu Aug 23 15:35:46 2014` |
+/// | `{C}`, `{year-short}` | Short year | `22`, `20` |
+/// | `{Y}`, `{year}` | Year | `2022`, `2021` |
+/// | `{D}`, `{date-short}` | Short date | `04/01/22`, `12/31/21` |
+/// | `{m}`, `{month}` | Month | `01`, `12` |
+/// | `{d}`, `{day}` | Day in month | `01`, `12`, `31`, `30` |
+/// | `{H}`, `{hour}` | Hour in 24-hour | `01`, `12`, `23` |
+/// | `{I}`, `{hour-12}` | Hour in 12-hour | `01`, `12` |
+/// | `{M}`, `{minute}` | Minute | `00`, `05`, `59` |
+/// | `{S}`, `{second}` | Second | `00`, `05`, `59` |
+/// | `{e}`, `{millisecond}` | Millisecond | `231` |
+/// | `{f}`, `{microsecond}` | Microseconds within a second | `372152` |
+/// | `{F}`, `{nanosecond}` | Nanoseconds within a second | `482930154` |
+/// | `{p}`, `{ampm}` | AM / PM | `AM`, `PM` |
+/// | `{r}`, `{time-12}` | Time in 12-hour format | `02:55:02 PM` |
+/// | `{R}`, `{time-short}` | Short time | `22:28`, `09:53` |
+/// | `{T}`, `{X}`, `{time}` | Time | `22:28:02`, `09:53:41` |
+/// | `{z}`, `{tz-offset}` | Timezone offset | `+08:00`, `+00:00`, `-06:00` |
+/// | `{E}`, `{unix-timestamp}` | Unix timestamp | `1528834770` |
+/// | `{+}`, `{full}` | Full log message | See [`FullFormatter`] |
+/// | `{l}`, `{level}` | Log level | `critical`, `error`, `warn` |
+/// | `{L}`, `{level-short}` | Short log level | `C`, `E`, `W` |
+/// | `{@}`, `{loc}` | Log location | `main.rs:30:20` |
+/// | `{s}`, `{source-basename}` | Source file basename | `main.rs` |
+/// | `{g}`, `{source}` | Path to the source file | `src/main.rs` |
+/// | `{#}`, `{line}` | Source file line | `30` |
+/// | `{%}`, `{column}` | Source file column | `20` |
+/// | `{n}`, `{logger}` | Logger name | `my-logger` |
+/// | `{v}`, `{payload}` | Log payload | `log message` |
+/// | `{P}`, `{pid}` | Process ID | `3824` |
+/// | `{t}`, `{tid}` | Thread ID | `3132` |
+///
+/// [`FullFormatter`]: crate::formatter::FullFormatter
+#[macro_export]
+macro_rules! pattern {
+    ( $($t:tt)* ) => {
+        $crate::formatter::macros::pattern_impl!($($t)*)
+    }
+}
+
+/// Build a [`PatternFormatter`] from a pattern built by the [`pattern`] macro
+/// with the given macro arguments.
+///
+/// `pattern_formatter!(...)` is equivalent to
+/// `PatternFormatter::new(pattern!(...))`.
+///
+/// ```
+/// # use spdlog::pattern_formatter;
+/// # use spdlog::formatter::PatternFormatter;
+/// #
+/// let formatter: PatternFormatter<_> = pattern_formatter!("{n}: {^[{level}]$} {v}");
+/// ```
+#[macro_export]
+macro_rules! pattern_formatter {
+    ( $($t:tt)* ) => {
+        $crate::formatter::PatternFormatter::new($crate::pattern!($($t)*))
+    };
+}
```

Comment:
This macro provides a convenient way of using patterns since in most cases, users are using:

```rust
PatternFormatter::new(pattern!(/*...*/))
```